sdcm/nemesis.py

```diff
@@ -37,6 +38,60 @@
 from sdcm.utils import remote_get_file
 
 
+def run_in_mixed_mode(NemesisCls):
+    """Run Nemesis in mixed mode on aws deployment
+
+    if aws deployment set with mixed mode, when using
+    Spot and OnDemand instances, the Nemesis which start,
+    stop, restart nemesis, will be run on OnDemand instances.
+
+    Function is a decorator, which is used for appropriate
+    Nemesis class
+
+
+    Arguments:
+        NemesisCls {class} -- Class of Nemesis to decorate
+
+    Returns:
+        [class] -- decorated Nemesis Class with rewritten methods
+    """
+    class MixedModeNemesis(NemesisCls):
+        """Decorator class
+
+        Inherit the original Nemesis and rewrite methods
+
+        Extends:
+            NemesisCls
+        """
+        def set_target_node(self):
+            if (self.tester.params.get('instance_provision') == 'mixed' and
```

Comment:
Why do we need to check if it's mixed mode?

Maybe we can simplify it by just building the list of the non_seed_nodes and choosing from that list.
If the list is empty, skip the nemesis.

Reply:
If we choose to do that, we can change the decorator name to "not_run_on_spot" or something like this.

Reply:
As i understood the task, this functionality should work only in mixed mode, when it is set in config file. in all other modes (provision_instances != mixed), it should work as before.
If such nemesises, should not run in any mode of provision instance, then this check is not required, you are right

Reply:
Adding one moe decorator. It allow mark appropriate disrupt method. And allow not to run disrupt method on spots in any mode. in this case there is no need to decorate the Child NemesisClasses and such nemesis as Choas and relative, will run seperatly on instance provision modes